src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -68,6 +69,16 @@
     @Query("select distinct participation from StudentParticipation participation left join fetch participation.results result where participation.exercise.id = :#{#exerciseId} and (result.id = (select max(id) from participation.results) or result is null)")
     List<StudentParticipation> findByExerciseIdWithLatestResult(@Param("exerciseId") Long exerciseId);
 
+    /**
+     * Get all participations for an exercise with each latest {@link AssessmentType#AUTOMATIC} result (determined by id).
+     * If there is no latest result (= no result at all), the participation will still be included in the returned ResultSet, but will have an empty Result array.
+     *
+     * @param exerciseId Exercise id.
+     * @return participations for exercise.
+     */
+    @Query("select distinct participation from StudentParticipation participation left join fetch participation.results result where participation.exercise.id = :#{#exerciseId} and (result.id = (select max(prs.id) from participation.results prs where prs.assessmentType = 'AUTOMATIC') or result is null)")
```

Comment:
do we really need the part `or result is null`?
We cannot update a non existing result so we don't even need to fetch it from the database right?

Reply:
I adapted that from the query above, but I guess we can also leave it out.

Reply:
I would leave it out for performance reasons. `Or` is typically not performant and we would filter those participations out later anyway